demisto_sdk/commands/content_graph/objects/correlation_rule.py

```diff
@@ -7,6 +10,9 @@
 
 
 class CorrelationRule(ContentItemXSIAM, content_type=ContentType.CORRELATION_RULE):  # type: ignore[call-arg]
+    execution_mode: Optional[str] = Field(None)
```

Comment:
Let's make this field an enum with possible values 'REAL_TIME', 'SCHEDULED'.
You can see PlaybookTaskType in base_playbook object as a reference.
Make sure to fix the validation and test accordingly.